### PR TITLE
Implement Pre-Fill Functionality for Forms 21-22 & 21-22a

### PR DESCRIPTION
**Pull request by cosu419**
## Summary
Added prefill classes, profile mappings for 21-22 & 21-22a.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/83555

## Testing done
- Added rspec covering each form prefill

## What areas of the site does it impact?
Appoint a representative app (not yet in production)

### CODE REVIEW

**Review by opticbob — DISMISSED** [review]
Looks good to me!  

I worked off of the [documentation](https://depo-platform-documentation.scrollhelp.site/developer-docs/va-forms-library-how-to-work-with-pre-fill#VAFormsLibrary-HowtoworkwithPre-Fill-Vets-APIWork) and it looks like you hit all the highlights!

File: spec/models/form_profile_spec.rb [hunk 1]
```diff
@@ -866,6 +866,48 @@
     }
   end
 
+  let(:v21_22_expected_) do
```
Comment (holdenhinkle) [inline spec/models/form_profile_spec.rb:869]: Delete the trailing `_` in `v21_22_expected_`.

Also, where is this being used?

File: spec/models/form_profile_spec.rb [hunk 2]
```diff
@@ -866,6 +866,48 @@
...
+      'contactInformation' => {
+        'email' => user.pciu_email,
+        'phone' => us_phone,
+        'address' => address
+      },
+      'veteranServiceInformation' => veteran_service_information
+    }
+  end
+
+  let(:v21_22a_expected_) do
```
Comment (holdenhinkle) [inline spec/models/form_profile_spec.rb:890]: Delete the trailing `_` in `v21_22a_expected_`.

Also, where is this being used?

Comment (rmtolmach) [inline spec/models/form_profile_spec.rb:890]: Does any test call these expected blocks? I think you'll need to write a test for them, or add these forms to other tests like this: https://github.com/department-of-veterans-affairs/vets-api/blob/master/spec/models/form_profile_spec.rb#L1605-L1628

Also, I noticed there was an underscore at the end of the `expected` block names. Is this intentional?

File: app/models/form_profiles/va_2122a.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class FormProfiles::VA2122a < FormProfile
+  def metadata
+    {
+      version: 0,
+      prefill: true,
+      returnUrl: '/claimant-personal-information'
```
Comment (rmtolmach) [inline app/models/form_profiles/va_2122a.rb:8]: How have you tested that `'/claimant-personal-information'` is valid?

File: spec/models/form_profile_spec.rb (hunk 2 above)
Comment (rmtolmach) [reply spec/models/form_profile_spec.rb]: Holden and I have the same thoughts today 😃

Comment (cosu419) [reply spec/models/form_profile_spec.rb]: Added 21_22 and 21_22A to the existing prefill test [here](https://github.com/department-of-veterans-affairs/vets-api/pull/17598/commits/a514c4cbd70a5e8fcb6f1e27447352885fe80d8e).

File: app/models/form_profiles/va_2122a.rb (hunk 3 above)
Comment (cosu419) [reply app/models/form_profiles/va_2122a.rb:8]: Hi @rmtolmach - is the `returnUrl` being tested for any of the other form profiles? I didn't immediately find any examples of that.

**Review by holdenhinkle — APPROVED** [review]

**Review by holdenhinkle — APPROVED** [review]
LGTM!

File: app/models/form_profiles/va_2122a.rb (hunk 3 above)
Comment (RachalCassity) [reply app/models/form_profiles/va_2122a.rb:8]: @cosu419 According to the docs, in this issue, the returnUrl is '/claimant-information'

Comment (RachalCassity) [reply app/models/form_profiles/va_2122a.rb:8]: Screenshot 2024-08-09 at 3 26 33 PM

Comment (RachalCassity) [reply app/models/form_profiles/va_2122a.rb:8]: I also see `/claimant-contact-information`. Can you confirm this is a valid url?

Comment (cosu419) [reply app/models/form_profiles/va_2122a.rb]: It looks like `/claimant-information` would be appropriate here. Updated [here](https://github.com/department-of-veterans-affairs/vets-api/pull/17598/commits/c74b7a0a283e796f8e0a3604d99988d86958a343).

**Review by RachalCassity — APPROVED** [review]

**Pull request merged**